mp/src/game/shared/momentum/weapon/weapon_mom_stickybomblauncher.cpp

```diff
@@ -0,0 +1,436 @@
+#include "cbase.h"
+
+#include "in_buttons.h"
+#include "mom_player_shared.h"
+#include "mom_stickybomb.h"
+#include "mom_system_gamemode.h"
+#include "weapon_mom_stickybomblauncher.h"
+
+#ifdef CLIENT_DLL
+#include "prediction.h"
+#else
+#include "momentum/mom_triggers.h"
+#endif
+
+#include "tier0/memdbgon.h"
+
+#define MOM_STICKYBOMB_MIN_CHARGE_VEL 900
+#define MOM_STICKYBOMB_MAX_CHARGE_VEL 2400
+#define MOM_STICKYBOMB_MAX_CHARGE_TIME 4.0f
+#define MOM_WEAPON_STICKYBOMB_COUNT 3
+
+IMPLEMENT_NETWORKCLASS_ALIASED(MomentumStickybombLauncher, DT_MomentumStickybombLauncher)
+
+BEGIN_NETWORK_TABLE_NOBASE(CMomentumStickybombLauncher, DT_MomentumStickybombLauncherLocalData)
+#ifdef CLIENT_DLL
+    RecvPropInt(RECVINFO(m_iStickybombCount)),
+    RecvPropFloat(RECVINFO(m_flChargeBeginTime)),
+    RecvPropBool(RECVINFO(m_bIsChargeEnabled)),
+#else
+    SendPropInt(SENDINFO(m_iStickybombCount), 5, SPROP_UNSIGNED),
+    SendPropFloat(SENDINFO(m_flChargeBeginTime), 5, SPROP_NOSCALE | SPROP_CHANGES_OFTEN),
+    SendPropBool(SENDINFO(m_bIsChargeEnabled)),
+#endif
+END_NETWORK_TABLE()
+
+BEGIN_NETWORK_TABLE(CMomentumStickybombLauncher, DT_MomentumStickybombLauncher)
+#ifdef CLIENT_DLL
+    RecvPropDataTable("MomentumStickybombLauncherLocalData", 0, 0, &REFERENCE_RECV_TABLE(DT_MomentumStickybombLauncherLocalData)),
+#else
+    SendPropDataTable("MomentumStickybombLauncherLocalData", 0, &REFERENCE_SEND_TABLE(DT_MomentumStickybombLauncherLocalData), SendProxy_SendLocalWeaponDataTable),
+#endif
+END_NETWORK_TABLE()
+
+#ifdef CLIENT_DLL
+BEGIN_PREDICTION_DATA(CMomentumStickybombLauncher) 
+    DEFINE_FIELD(m_flChargeBeginTime, FIELD_FLOAT)
+END_PREDICTION_DATA()
+#endif
+
+LINK_ENTITY_TO_CLASS(weapon_momentum_stickylauncher, CMomentumStickybombLauncher);
+PRECACHE_WEAPON_REGISTER(weapon_momentum_stickylauncher);
+
+#ifdef GAME_DLL
+static MAKE_TOGGLE_CONVAR_CV(mom_sj_firing_mode, "1", FCVAR_ARCHIVE,
```

Comment:
I'm not sure I'm digging the new name. Every instance I've seen the var used is only `GetBool` but I've had to look back at the comment of the convar to determine which mode is which (ambiguous name). It would be worth it to go back to `mom_sj_charge_enable`.